staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -399,17 +410,60 @@ func (c completedConfig) New() (*GenericAPIServer, error) {
 
 		apiGroupsForDiscovery: map[string]metav1.APIGroup{},
 
+		HandlerContainer:   handlerContainer,
 		FallThroughHandler: c.FallThroughHandler,
 
+		listedPathProvider: routes.ListedPathProviders{handlerContainer, c.FallThroughHandler},
+
 		swaggerConfig: c.SwaggerConfig,
 		openAPIConfig: c.OpenAPIConfig,
 
 		postStartHooks: map[string]postStartHookEntry{},
 		healthzChecks:  c.HealthzChecks,
 	}
 
-	s.HandlerContainer = mux.NewAPIContainer(http.NewServeMux(), c.Serializer, s.FallThroughHandler)
+	return s, nil
+}
+
+// Combine creates a new server which logically combines the handling chain with the passed server.
+func (c completedConfig) Combine(delegationTarget DelegationTarget) (*GenericAPIServer, error) {
```

Comment:
> Having tpr-apiserver in mind some day.

I share the vision.  This gets us a lot closer, but I can't say that it's exactly what we want for that future time.